include/nbl/builtin/hlsl/emulated_float64_t.hlsl

```diff
@@ -0,0 +1,596 @@
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+
+#ifdef __HLSL_VERSION
+#define LERP lerp
+#else
+#define LERP nbl::hlsl::lerp
+#endif
+
+#ifdef __HLSL_VERSION
+#define ABS abs
+#else
+#define ABS std::abs
+#endif
+
+// TODO: inline function
+#define EXCHANGE(a, b) \
+   do {                \
+       a ^= b;         \
+       b ^= a;         \
+       a ^= b;         \
+   } while (false)
+   
+#define FLOAT_ROUND_NEAREST_EVEN    0
+#define FLOAT_ROUND_TO_ZERO         1
+#define FLOAT_ROUND_DOWN            2
+#define FLOAT_ROUND_UP              3
+#define FLOAT_ROUNDING_MODE         FLOAT_ROUND_NEAREST_EVEN

```

Comment:
remember to `#undef` all this crap after the end of the header.

btw the `abs` and `lerp` should probably both be forwarded to `nbl::hlsl::` in `cpp_compat.hlsl` or one of th headers it includes

Reply:
as we agreed, implemented only round to zero